### ISSUE TITLE
Checker interface is out of date

### ISSUE THREAD

**Issue by apblack** [gracelang/minigrace#178]
We have agreed that the interface of a dialect should look like this:

``` grace
def thisDialect is public = object {
    method parseChecker (moduleObj) { check (moduleObj) }
    method astChecker (moduleObj) { check (moduleObj) }
    method atStart (moduleName:String) { print "module start" }
    method atEnd (moduleObj) { print "module end" }
}
```

(This example comes from _modules/staticTypes.grace_).   At present, this is not implemented; instead, _xmodule_ looks for top-level methods called `checker` and `astChecker`.

This should be fixed.

**Comment by petercowal** [comment]
I've updated the dialect interface as follows:
```grace
def thisDialect is public = object {

    method parseChecker (moduleNode) { 
         // call parsing checks here
    }
    method astChecker (moduleNode) { 
        // call AST checks here
    }
    method atStart (moduleName:String) { 
        // this will run before the user's code is executed
    }
    method atEnd (moduleObj) {
        // this will run after the user's code is executed
    }
}
```

**Comment by apblack** [comment]
This was fixed in commit fe53510

**Comment by apblack** [comment]
This interface is now documented on the documentation website under "Writing a dialect"
